#! /bin/sh -e 
## clean.dpatch by <luther@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Makefiles, configure scripts and other build stuff adapted for
## DP: debian package creation

if [ $# -ne 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;	
    *)
	echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
	exit 1;;
esac

exit 0
@DPATCH@
diff -ur ocaml-3.07beta2.orig/otherlibs/labltk/Makefile ocaml-3.07beta2/otherlibs/labltk/Makefile
--- ocaml-3.07beta2.orig/otherlibs/labltk/Makefile	2002-05-16 17:23:05.000000000 +0200
+++ ocaml-3.07beta2/otherlibs/labltk/Makefile	2003-08-26 14:45:32.000000000 +0200
@@ -3,7 +3,7 @@
 SUBDIRS=compiler support lib jpf frx tkanim examples_labltk \
 	camltk labltk examples_camltk browser
 
-all:
+all: backup
 	cd support; $(MAKE)
 	cd compiler; $(MAKE)
 	cd labltk; $(MAKE) -f Makefile.gen
@@ -16,7 +16,7 @@
 	cd tkanim; $(MAKE)
 	cd browser; $(MAKE)
 
-allopt:
+allopt: backup
 	cd support; $(MAKE) opt
 	cd labltk; $(MAKE) -f Makefile.gen
 	cd labltk; $(MAKE) opt
@@ -71,5 +71,13 @@
 	for d in $(SUBDIRS); do \
 	    cd $$d; $(MAKE) -f Makefile clean; cd ..; \
 	done
+	-mv README.labltk README
+	-mv labltk/.depend.original labltk/.depend
+	-mv camltk/.depend.original camltk/.depend
+
+backup:
+	-mv README README.labltk
+	-cp labltk/.depend labltk/.depend.original
+	-cp camltk/.depend camltk/.depend.original
 
 depend:
